website/docs/r/waf_rule.html.markdown

```diff
@@ -54,7 +54,7 @@ The following arguments are supported:
   For example, if an IPSet includes the IP address `192.0.2.44`, AWS WAF will allow or block requests based on that IP address.
   If set to `true`, AWS WAF will allow, block, or count requests based on all IP addresses _except_ `192.0.2.44`.
 * `data_id` - (Required) A unique identifier for a predicate in the rule, such as Byte Match Set ID or IPSet ID.
-* `type` - (Required) The type of predicate in a rule, such as `ByteMatchSet` or `IPSet`
+* `type` - (Required) The type of predicate in a rule. Must be one of `IPMatch`, `ByteMatch`, `SqlInjectionMatch`, `SizeConstraint`, `XssMatch`, or `GeoMatch`.
```

Comment:
Nitpick: We don't usually try to put extensive list of all valid values in our documentation, but instead link to official docs. This is to avoid a constant uphill battle of being out of date as Amazon releases new stuff and changes these things.